dashboard/src/components/partials/EventDetailsContainer/EventDetailsResolveButton.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import PropTypes from "prop-types";
+import gql from "graphql-tag";
+import Button from "material-ui/Button";
+import resolveEvent from "/mutations/resolveEvent";
+
+class EventDetailsResolveButton extends React.PureComponent {
+  static propTypes = {
+    client: PropTypes.object.isRequired,
+    disabled: PropTypes.bool,
+    event: PropTypes.object,
+    onProcessing: PropTypes.func.isRequired,
```

Comment:
This `onProcessing` interface feels fragile.

Reply:
Maybe a set of callbacks: `onResolveEnd` and `onResolveFail`, would help decouple this from the actual locking state upstream.